pkg/pubsub/dapr/dapr.go

```diff
@@ -0,0 +1,100 @@
+package dapr
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/dapr/go-sdk/client"
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/common"
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/connection"
+)
+
+type ClientConfig struct {
+	// Enable, Disable batching per tool
+	EnableBatching bool `json:"enableBatching,omitempty"`
+
+	// batch size
+	Size int `json:"size"`
+
+	// Name of the component to be used for pub sub messaging
+	Component string `json:"component"`
+}
+
+// Dapr represents driver for interacting with pub sub using dapr.
+type Dapr struct {
+	// Array of clients to talk to different endpoints
+	client client.Client
+
+	// Name of the pubsub component
+	pubSubComponent string
+
+	// Enable, Disable batching per tool
+	batchingEnabled bool
+
+	// batch size
+	size int
+}
+
+const (
+	Name          = "dapr"
+	componentName = "pubsub"
+)
+
+func (r *Dapr) Publish(ctx context.Context, data interface{}, topic string) error {
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		return fmt.Errorf("error marshaling data: %w", err)
+	}
+
+	err = r.client.PublishEvent(context.Background(), r.pubSubComponent, topic, jsonData)
+	if err != nil {
+		return fmt.Errorf("error publishing message to dapr: %w", err)
+	}
+
+	return nil
+}
+
+func (r *Dapr) PublishBatch(ctx context.Context, data interface{}, topic string) error {
+	// add logic of batch publishing with dapr from 1.10
```

Comment:
is this a follow up or do we want to do this in this pr?

Reply:
Part of this PR. I am working on this and understanding dapr's changes that are released in 1.10

Reply:
@sozercan @maxsmythe @ritazh Sounds like changes to dapr go-sdk to support bulk publish and subscribe are yet to be implemented - [here](https://github.com/dapr/go-sdk/issues/325). I think they are going SDK by SDK, cause js SDK has this support as of now. How do you think we should move forward? As I see it we have two options:
1. Acknowledge that as of now BatchPublish will not work with gatekeeper as go-sdk lacks the support and follow-up to update the driver whenever bulk publish support is added to go SDK
2. We create a go channel to call publish from within `batchPublish` and imitate batching (I think this is what you mean @maxsmythe in [this comment](https://github.com/open-policy-agent/gatekeeper/pull/2538#discussion_r1106704543)).

Reply:
I think we're on the same page with (2), which seems the right approach: dapr doesn't support batch publishing currently, but will in the future, therefore the driver polyfills batch publishing, but won't in the future.

Reply:
Cool! Implemented `PublishBatch` as we discussed.

Reply:
This code should not live in the driver, but in the system.